### PR TITLE
Refactor use TrimSuffix instead of TrimRight

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

File: models/ssh_key.go [hunk 1]
```diff
@@ -318,7 +318,7 @@ func CheckPublicKeyString(content string) (_ string, err error) {
 		return "", err
 	}
 
-	content = strings.TrimRight(content, "\n\r")
+	content = strings.TrimSuffix(content, "\n\r")
```
Comment (6543) [inline models/ssh_key.go]:
```suggestion
	content = strings.TrimRight(content, "\n\r")
```

File: modules/git/repo_tag.go [hunk 2]
```diff
@@ -193,7 +193,7 @@ func (repo *Repository) GetTagInfos(page, pageSize int) ([]*Tag, error) {
 		return nil, err
 	}
 
-	tagNames := strings.Split(strings.TrimRight(stdout, "\n"), "\n")
+	tagNames := strings.Split(strings.TrimSuffix(stdout, "\n"), "\n")
```
Comment (6543) [inline modules/git/repo_tag.go]:
```suggestion
	tagNames := strings.Split(strings.TrimRight(stdout, "\n"), "\n")
```

File: modules/git/tag.go [hunk 3]
```diff
@@ -60,7 +60,7 @@ l:
 			}
 			nextline += eol + 1
 		case eol == 0:
-			tag.Message = strings.TrimRight(string(data[nextline+1:]), "\n")
+			tag.Message = strings.TrimSuffix(string(data[nextline+1:]), "\n")
```
Comment (6543) [inline modules/git/tag.go]:
```suggestion
			tag.Message = strings.TrimRight(string(data[nextline+1:]), "\n")
```

File: modules/public/public.go [hunk 4]
```diff
@@ -79,7 +79,7 @@ func (opts *Options) staticHandler(dir string) macaron.Handler {
 			opts.Prefix = "/" + opts.Prefix
 		}
 		// Remove any trailing '/'
-		opts.Prefix = strings.TrimRight(opts.Prefix, "/")
+		opts.Prefix = strings.TrimSuffix(opts.Prefix, "/")
```
Comment (6543) [inline modules/public/public.go]:
```suggestion
		opts.Prefix = strings.TrimRight(opts.Prefix, "/")
```

File: modules/webhook/telegram.go [hunk 5]
```diff
@@ -120,7 +120,7 @@ func (t *TelegramPayload) Push(p *api.PushPayload) (api.Payloader, error) {
 			authorName = " - " + commit.Author.Name
 		}
 		text += fmt.Sprintf(`[<a href="%s">%s</a>] %s`, commit.URL, commit.ID[:7],
-			strings.TrimRight(commit.Message, "\r\n")) + authorName
+			strings.TrimSuffix(commit.Message, "\r\n")) + authorName
```
Comment (6543) [inline modules/webhook/telegram.go]:
```suggestion
			strings.TrimRight(commit.Message, "\r\n")) + authorName
```

File: modules/webhook/msteams.go [hunk 6]
```diff
@@ -226,7 +226,7 @@ func (m *MSTeamsPayload) Push(p *api.PushPayload) (api.Payloader, error) {
 	// for each commit, generate attachment text
 	for i, commit := range p.Commits {
 		text += fmt.Sprintf("[%s](%s) %s - %s", commit.ID[:7], commit.URL,
-			strings.TrimRight(commit.Message, "\r\n"), commit.Author.Name)
+			strings.TrimSuffix(commit.Message, "\r\n"), commit.Author.Name)
```
Comment (6543) [inline modules/webhook/msteams.go]:
```suggestion
			strings.TrimRight(commit.Message, "\r\n"), commit.Author.Name)
```

File: modules/webhook/feishu.go [hunk 7]
```diff
@@ -89,7 +89,7 @@ func (f *FeishuPayload) Push(p *api.PushPayload) (api.Payloader, error) {
 			authorName = " - " + commit.Author.Name
 		}
 		text += fmt.Sprintf("[%s](%s) %s", commit.ID[:7], commit.URL,
-			strings.TrimRight(commit.Message, "\r\n")) + authorName
+			strings.TrimSuffix(commit.Message, "\r\n")) + authorName
```
Comment (6543) [inline modules/webhook/feishu.go]:
```suggestion
			strings.TrimRight(commit.Message, "\r\n")) + authorName
```

File: modules/webhook/discord.go [hunk 8]
```diff
@@ -207,7 +207,7 @@ func (d *DiscordPayload) Push(p *api.PushPayload) (api.Payloader, error) {
 	// for each commit, generate attachment text
 	for i, commit := range p.Commits {
 		text += fmt.Sprintf("[%s](%s) %s - %s", commit.ID[:7], commit.URL,
-			strings.TrimRight(commit.Message, "\r\n"), commit.Author.Name)
+			strings.TrimSuffix(commit.Message, "\r\n"), commit.Author.Name)
```
Comment (6543) [inline modules/webhook/discord.go]:
```suggestion
			strings.TrimRight(commit.Message, "\r\n"), commit.Author.Name)
```

File: modules/webhook/dingtalk.go [hunk 9]
```diff
@@ -120,7 +120,7 @@ func (d *DingtalkPayload) Push(p *api.PushPayload) (api.Payloader, error) {
 			authorName = " - " + commit.Author.Name
 		}
 		text += fmt.Sprintf("[%s](%s) %s", commit.ID[:7], commit.URL,
-			strings.TrimRight(commit.Message, "\r\n")) + authorName
+			strings.TrimSuffix(commit.Message, "\r\n")) + authorName
```
Comment (6543) [inline modules/webhook/dingtalk.go]:
```suggestion
			strings.TrimRight(commit.Message, "\r\n")) + authorName
```

File: modules/setting/setting.go [hunk 10]
```diff
@@ -933,8 +933,8 @@ func NewContext() {
 	newMarkup()
 
 	sec = Cfg.Section("U2F")
-	U2F.TrustedFacets, _ = shellquote.Split(sec.Key("TRUSTED_FACETS").MustString(strings.TrimRight(AppURL, "/")))
```
Comment (techknowlogick) [inline modules/setting/setting.go:936]: Can you keep these two lines as previous otherwise it'll conflict with my PR which has these changes already.

**Review by lunny — APPROVED** [review]

File: modules/setting/setting.go [hunk 11]
```diff
@@ -933,8 +933,8 @@ func NewContext() {
...
-	U2F.AppID = sec.Key("APP_ID").MustString(strings.TrimRight(AppURL, "/"))
+	U2F.TrustedFacets, _ = shellquote.Split(sec.Key("TRUSTED_FACETS").MustString(strings.TrimSuffix(AppURL, "/")))
```
Comment (6543) [inline modules/setting/setting.go:936]:
```suggestion
	U2F.TrustedFacets, _ = shellquote.Split(sec.Key("TRUSTED_FACETS").MustString(strings.TrimRight(AppURL, "/")))
```

File: modules/setting/setting.go [hunk 12]
```diff
@@ -933,8 +933,8 @@ func NewContext() {
...
+	U2F.AppID = sec.Key("APP_ID").MustString(strings.TrimSuffix(AppURL, "/"))
```
Comment (6543) [inline modules/setting/setting.go:937]:
```suggestion
	U2F.AppID = sec.Key("APP_ID").MustString(strings.TrimRight(AppURL, "/"))
```

File: modules/setting/setting.go (hunk 11 above)
Comment (6543) [inline modules/setting/setting.go]:
```suggestion
	U2F.TrustedFacets, _ = shellquote.Split(sec.Key("TRUSTED_FACETS").MustString(strings.TrimRight(AppURL, "/")))
```

**Comment by 6543** [comment]
@techknowlogick done

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**